src/interface/sparse_pack_base.cpp

```diff
@@ -246,6 +244,10 @@ std::string SparsePackCache::GetIdentifier(const PackDescriptor &desc) const {
   identifier += "____";
   for (int i = 0; i < desc.vars.size(); ++i)
     identifier += desc.vars[i] + std::to_string(desc.use_regex[i]);
+  identifier += "____";
+  identifier += std::to_string(desc.with_fluxes);
+  identifier += std::to_string(desc.coarse);
+  identifier += std::to_string(desc.flat);
```

Comment:
This is definitely better than rebuilding every time if there are coarse and fine packs with the same set of variables, for instance.